src/Features/VisualBasic/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.vb

```diff
@@ -0,0 +1,37 @@
+﻿Imports System.Composition
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory
+
+<ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeRefactoringProviderNames.ExtractMethod), [Shared]>
+Partial Friend Class ConvertToInterpolatedStringRefactoringProvider
+    Inherits AbstractConvertToInterpolatedStringRefactoringProvider(Of InterpolatedStringExpressionSyntax, InterpolatedStringContentSyntax, InterpolationSyntax, InvocationExpressionSyntax, ExpressionSyntax, ArgumentSyntax, LiteralExpressionSyntax)
+
+    Protected Overrides Function GetArguments(invocation As InvocationExpressionSyntax) As SeparatedSyntaxList(Of ArgumentSyntax) ?
+        Return invocation?.ArgumentList.Arguments
+    End Function
+
+    Protected Overrides Function GetInterpolatedString(text As String) As InterpolatedStringExpressionSyntax
+        Return TryCast(ParseExpression("$" + text), InterpolatedStringExpressionSyntax)
+    End Function
+
+    Protected Overrides Function GetText(literalExpression As LiteralExpressionSyntax) As String
+        ' We need to escape braces as this is an ambiguous case in interpolated strings that is 
+        ' not ambiguous in verbatim strings
+        Return literalExpression.Token.ToString.Replace("'{'", "'{{'").Replace("'}'", "'}}'")
```

Comment:
but its not.  All strings are verbatim in VB so we must always do this replacement.  In C# only strings with `@` are verbatim.  Because the semantics are different here I think this is one place where we should have separate methods, but I could combine their common logic in some way...